merlin/core/dispatch.py

```diff
@@ -24,16 +24,16 @@
 import pyarrow as pa
 import pyarrow.parquet as pq
 
-from merlin.core.compat import HAS_GPU
+# unused HAS_GPU import is here for backwards compatibility
+from merlin.core.compat import HAS_GPU  # pylint: disable=unused-import # noqa: F401
+from merlin.core.compat import cupy as cp
 from merlin.core.protocols import DataFrameLike, DictLike, SeriesLike
 
 cudf = None
-cp = None
 rmm = None
 
-if HAS_GPU:
+if cudf:
```

Comment:
I could be missing something, but seems like `cudf` would need to imported from `merlin.core.compat` for this to work

Reply:
I missed this commit from moving from the other PR. Updated